nomos-services/data-availability/tests/src/verifier_integration.rs

```diff
@@ -0,0 +1,301 @@
+        &coins[0],
+        &ledger_config,
+        &genesis_state,
+        &time_config,
+        &swarm_config1,
+        NamedTempFile::new().unwrap().path().to_path_buf(),
+        &blobs_dir,
+        vec![node_address(&swarm_config2)],
+        KzgrsDaVerifierSettings {
+            sk: node1_sk,
+            nodes_public_keys: vec![node1_pk, node2_pk],
+        },
+    );
+
+    let node2 = new_node(
+        &coins[1],
+        &ledger_config,
+        &genesis_state,
+        &time_config,
+        &swarm_config2,
+        NamedTempFile::new().unwrap().path().to_path_buf(),
+        &blobs_dir,
+        vec![node_address(&swarm_config1)],
+        KzgrsDaVerifierSettings {
+            sk: node2_sk,
+            nodes_public_keys: vec![node1_pk, node2_pk],
+        },
+    );
+
+    let node1_verifier = node1.handle().relay::<DaVerifier>();
+
+    let node2_verifier = node2.handle().relay::<DaVerifier>();
+
+    client_zone.spawn(async move {
+        let node1_verifier = node1_verifier.connect().await.unwrap();
+        let (node1_reply_tx, node1_reply_rx) = tokio::sync::oneshot::channel();
+
+        let node2_verifier = node2_verifier.connect().await.unwrap();
+        let (node2_reply_tx, node2_reply_rx) = tokio::sync::oneshot::channel();
+
+        let verifiers = vec![
+            (node1_verifier, node1_reply_tx),
+            (node2_verifier, node2_reply_tx),
+        ];
+
+        // Encode data
+        let encoder = &ENCODER;
+        let data = rand_data(10);
+
+        let encoded_data = encoder.encode(&data).unwrap();
+        let columns: Vec<_> = encoded_data.extended_data.columns().collect();
+
+        for (i, (verifier, reply_tx)) in verifiers.into_iter().enumerate() {
+            let column = &columns[i];
+
+            let da_blob = DaBlob {
+                column: column.clone(),
+                column_commitment: encoded_data.column_commitments[i],
+                aggregated_column_commitment: encoded_data.aggregated_column_commitment,
+                aggregated_column_proof: encoded_data.aggregated_column_proofs[i],
+                rows_commitments: encoded_data.row_commitments.clone(),
+                rows_proofs: encoded_data
+                    .rows_proofs
+                    .iter()
+                    .map(|proofs| proofs.get(i).cloned().unwrap())
+                    .collect(),
+            };
+
+            verifier
+                .send(nomos_da_verifier::DaVerifierMsg::AddBlob {
+                    blob: da_blob,
+                    reply_channel: reply_tx,
+                })
+                .await
+                .unwrap();
+        }
+
+        // Create cert
+        let a1 = node1_reply_rx.await.unwrap();
+        let a2 = node2_reply_rx.await.unwrap();
+
+        if a1.is_some() && a2.is_some() {
+            is_success_tx.store(true, SeqCst);
+        }
+
+        // TODO: Create cert and check indexer integration.
+        performed_tx.store(true, SeqCst);
```

Comment:
We will check with the indexer to see if the blob is available right? Gonna come in a different PR? (If so lets open an issue I think).